### PR TITLE
Replace gdp with Go 1.6+ built-in vendoring, update all deps to latest

### PR DESCRIPTION
**Pull request by snwight**
Replace gdp with Go 1.6+ built-in vendoring. Updated all deps to latest versions, vendored them all so we have a snapshot of today's state. I used govendor to init and sync,  which is a sledgehammer approach and pulls in every last thing, which is what we probably want for this docker env. I made a single change to the source, in cmd/hologram-authorize/main.go to adapt for a recent change to gopass package. This builds and all tests - such as they are - pass under Go 1.7.1

### CODE REVIEW

**Comment by snwight** [comment]
Low hanging fruit, part deux... cleaned up header comments for packages so the 'Overview' section isn't filled with license info, added godoc generation target to buildscripts (**generate_docs.sh**) which serves up HTML godocs to localhost:6060 as well as static type analysis of source (so the source refs have live ref links, useful for navigation in the absence of a Go-sensitive editor env)

File: .idea/libraries/GOPATH__hologram_.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<component name="libraryTable">
+  <library name="GOPATH &lt;hologram&gt;">
+    <CLASSES>
+      <root url="file://$USER_HOME$/src/go/src/golang.org" />
+      <root url="file://$USER_HOME$/src/go/src/github.com" />
+    </CLASSES>
+    <SOURCES>
+      <root url="file://$USER_HOME$/src/go/src/golang.org" />
```
Comment (copumpkin) [inline .idea/libraries/GOPATH__hologram_.xml]: These seem rather user-specific, right?

Comment (snwight) [inline .idea/libraries/GOPATH__hologram_.xml]: just making me crazy - using jetbrains and i keep accidentally committing this crap... hopefully removed in next commit. thanks!

**Comment by snwight** [comment]
thanks for the observation, @copumpkin - i'm a recent convert to jetbrains IDE (from emacs) ... adding `.idea/` to .gitignore now

**Pull request merged**